### DISCUSSION TITLE
Does Eco2AI differentiate between consumption from different processes?

### DISCUSSION THREAD

**Discussion by adosar** [sb-ai-lab/Eco2AI#15] [General]
First of all, 👏 congratulations for the development of this unique and very practical package!

Apologize if this is a naive question, but does `Eco2AI` measures the consumption, e.g. CPU consumption, for each process individually or it sums the consumption from all processes?

From the [paper](https://arxiv.org/pdf/2208.00406):

> 
> Having much in common with recent analogs, in eco2AI we focused on the following: taking into
> account only those system processes that are related directly to models training (to avoid over-estimation);

I am a little bit confused if this is only for training or includes inference also.

As an example, suppose we train model A while we perform inference with model B, both in the same GPU device. Furthermore, we assume that training A is a lot more resource heavy that inference with model B.

Would `Eco2AI` report the same consumption for both cases, or it will report more consumption for A than B?

To put it differently, do we need to ensure that nothing else is running on the background so `Eco2AI` would return accurate estimates?
